netbox/extras/api/customfields.py

```diff
@@ -103,6 +103,9 @@ def _populate_custom_fields(instance, fields):
             for cfv in instance.custom_field_values.all():
                 if cfv.field.type == CF_TYPE_SELECT:
                     custom_fields[cfv.field.name] = CustomFieldChoiceSerializer(cfv.value).data
+                elif cfv.field.type == CF_TYPE_DATE:
```

Comment:
This shouldn't be necessary, as DRF should automatically serialize date objects. If this _is_ an issue, let's split it out from the middleware work since it's not really related.